### PR TITLE
Podfile injections

### PR DESCRIPTION
**Pull request by mihaiblaga89**
## Description

- added Podfile header support. Both apps and plugins can now add `templateXcode.Podfile.header` strings to be added on top of the Podfile
- added `enableUserDefinedBuildTypes` for ios platform. This enables user defined cocoapods build types (https://github.com/joncardasis/cocoapods-user-defined-build-types). However because the Gemfile is not part of the templated files we can't inject the gem dependency so that needs to be done manually before enabling this feature
- added pod buildType. It can be `static` or `dynamic`. Setting this along with the `podName` will allow RNV to inject it in podfile with that specific type, ex: `{ podName: "Firebase",  buildType: "static"}` => `pod 'Firebase', :build_type => :static_framework`
- pod injections do not happen automatically unless there's a reason to do so. Just having `podName` set will not inject it(no need, RN picks it up), it needs `buildType`, `commit`, `version` or `git` to do it. 
- updated defaults (deployment target, gradle version)

## TODO
- add harness testcases for each of this changes

## Related issues

- GH issues

## Npm releases

n/a

### CODE REVIEW

**Review by pavjacko — CHANGES_REQUESTED** [review]
this needs proper functioning test case in app-harness

**Comment by mihaiblaga89** [comment]
@pavjacko harness now has working: 
- sources injection
- header injection
- enableUserDefinedBuildTypes
- podnames injections
- custom pod git and build_type
example:

```
source 'https://github.com/CocoaPods/Specs.git'


$RNFirebaseAsStaticFramework = true
plugin 'cocoapods-user-defined-build-types' 
enable_user_defined_build_types!
------------
target 'RNVApp' do
  config = use_native_modules!

  pod 'Firebase', :build_type => :static_framework
pod 'FirebaseCoreInternal', :build_type => :static_framework
pod 'FirebaseCore', :build_type => :static_framework
pod 'GoogleUtilities', :build_type => :static_framework
pod 'FirebaseStorage', :build_type => :static_framework
pod 'FirebaseAppCheckInterop', :build_type => :static_framework
pod 'FirebaseAuthInterop', :build_type => :static_framework
  pod 'iOSPhotoEditor', :build_type => :dynamic_framework, :git => 'https://github.com/prscX/photo-editor', :commit => '4924e9ec984d25d03644e58aa148282642171de9'
```

File: packages/core/src/plugins/index.ts [hunk 1]
```diff
@@ -373,64 +373,70 @@ export const parsePlugins = (
...
             if (plugins) {
                 // let totalIncludedPlugins = 0;
                 Object.keys(plugins).forEach((key) => {
+                    const plugin = getMergedPlugin(c, key);
+
+                    if (!plugin) return;
+
                     if (
-                        (includedPlugins.includes('*') || includedPlugins.includes(key)) &&
+                        (includedPlugins!.includes('*') || includedPlugins!.includes(key)) &&
```
Comment (pavjacko) [inline packages/core/src/plugins/index.ts:411]: @mihaiblaga89 don't use non-null assertions

File: packages/engine-rn/templates/platforms/ios/Podfile [hunk 2]
```diff
@@ -1,3 +1,7 @@
+{{PLUGIN_PODFILE_SOURCES}}
+
+{{PODFILE_HEADER}}
```
Comment (pavjacko) [inline packages/engine-rn/templates/platforms/ios/Podfile:3]: @mihaiblaga89  prefix all injects with `INJECT_`

File: packages/core/src/schema/platforms/fragments/templateXcodeBase.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ const Podfile = z
         sources: z.optional(z.array(z.string())),
         podDependencies: z.optional(z.array(z.string())),
         staticPods: z.optional(z.array(z.string())),
+        header: z.optional(z.array(z.string())),
```
Comment (pavjacko) [inline packages/core/src/schema/platforms/fragments/templateXcodeBase.ts:10]: @mihaiblaga89 add .describe() to all new props

**Review by pavjacko — CHANGES_REQUESTED** [review]

File: packages/engine-rn/templates/platforms/ios/Podfile (hunk 2 above)
Comment (mihaiblaga89) [reply packages/engine-rn/templates/platforms/ios/Podfile]: switched to casting

**Review by pavjacko — APPROVED** [review]

**Pull request merged**